sql/core/src/main/scala/org/apache/spark/sql/parquet/newParquet.scala

```diff
@@ -630,11 +635,12 @@ object ParquetRelation2 {
             sqlContext.conf.isParquetBinaryAsString,
             sqlContext.conf.isParquetINT96AsTimestamp))
       }
-    }.reduce { (left, right) =>
-      try left.merge(right) catch { case e: Throwable =>
-        throw new SparkException(s"Failed to merge incompatible schemas $left and $right", e)
-      }
-    }
+    }.foldLeft[StructType](null) {
```

Comment:
All right. Instead of putting a large code block in `Option`, how about use a temporary `val` and then use `Option` at the end of this method.